crates/services/block_aggregator_api/src/blocks.rs

```diff
@@ -6,13 +6,18 @@ use std::fmt::Debug;
 /// Source from which blocks can be gathered for aggregation
 pub trait BlockSource: Send + Sync {
     /// Asynchronously fetch the next block and its height
-    fn next_block(&mut self)
-    -> impl Future<Output = Result<(BlockHeight, Block)>> + Send;
+    fn next_block(&mut self) -> impl Future<Output = Result<BlockSourceEvent>> + Send;
```

Comment:
i think this should be boxed, its a rather large type and i think it communicates intention better to someone reading the code